fs/driver/fs_blockpartition.c

```diff
@@ -224,6 +224,7 @@ static int part_geometry(FAR struct inode *inode, struct geometry *geometry)
 
 static int part_ioctl(FAR struct inode *inode, int cmd, unsigned long arg)
 {
+  FAR uintptr_t ptr_arg = (uintptr_t)arg;
```

Comment:
@xiaoxiang781216 I took care of this here because it is used in the AVR build, but we should probably do something like this in most of the ioctl that take a pointer as an argument.  This should be harmless since it is really just being explicit about the cast.  Just adding the cast everywhere in the function makes for long lines.